### ISSUE TITLE
[Feature Request] Configurable Snapclient pid file location

### ISSUE THREAD

**Issue by ghost** [badaix/snapcast#106]
I'm running Snapclient as a user service in Raspbian on my Raspberry PI 3. The `pi` user doesn't have write access to the `/var/run` directory, and `chmod`'ing it to work doesn't survive a reboot. This breaks the `-d` option for Snapclient (the exact error that it runs into is https://github.com/badaix/snapcast/blob/master/common/daemon.h#L80). It looks like #8 solves this for Snapserver, but doesn't change it for Snapclient. Would it be possible have a configurable pid file location for Snapclient as well?

I was able to fix the problem short-term by placing a file at `~/.config/autostart/snapclient.desktop` with the contents

``` desktop
[Desktop Entry]
Version=1.0
Name=Start Snapclient
Exec=/bin/sh -c "sleep 15; sudo chmod 777 /run; PULSE_LATENCY_MSEC=60 snapclient -d"
Terminal=false
Type=Application
```

**Comment by hell-g** [comment]
I have been trying to write a systemd user service for snapclient. I used the systemd service that comes with the Arch Linux snapcast package and adapted it. But no matter what PID file location I write in there, it always looks for the PID file at `/var/run/snapclient/pid`. Which I guess it cannot create as my user does not have permissions for this location. And therefore my user service fails to start. 
Is this PID file hardcoded? Could it be adapted, so it actually can be set in the systemd service file?
Thanks for looking into this!

**Comment by fuchsmich** [comment]
You don't need a PID file for systemd. If you don't run it as daemon, systemd takes care of everything.

https://build.merproject.org/package/view_file/home:fooxl:sailfish/snapcast/snapclient-nofork.service?expand=1

**Comment by hell-g** [comment]
@fuchsmich 
Thank you for your input!
Your first suggestion seems to work. I tested this snapclient.service file as user service:

> [Unit]
> Description=Snapcast client
> After=sound.target
> 
> [Service]
> EnvironmentFile=-/home/username/.snapcast/snapclient
> ExecStart=/usr/bin/snapclient $USER_OPTS $SNAPCLIENT_OPTS
> StandardOutput=null
> Restart=always
> 
> [Install]
> WantedBy=multi-user.target

I just copied over the /etc/default/snapclient file to my user's home directory. I had to remove the other `After` and `Requires` statements from your linked example, as they seem to be not available in a systemd user session. 

Your second suggestion does not work for me. As the service file includes `User=%i`, I get the same error message about the PID file:

> Exception: Could not open PID lock file "/var/run/snapclient/pid"

Thanks again for your help!
